mlrun/datastore/datastore.py

```diff
@@ -242,7 +242,9 @@ def get_or_create_store(
         if not secrets and not mlrun.config.is_running_as_api():
             self._stores[store_key] = store
         # in file stores in windows path like c:\a\b the drive letter is dropped from the path, so we return the url
-        return store, url if store.kind == "file" else subpath, url
+        if store.kind == "file" and isinstance(url, str):
+            subpath = url.replace("file://", "", 1)
```

Comment:
This fix is needed for when there is no schema in the URL (file://c:\a\b does not seem to work in windows)
I suggest you move the special-handling code above, and 
```
        if schema == "memory":
            subpath = url[len("memory://") :]
            return in_memory_store, subpath, url
        elif schema == "":
            # parse_url() will drop the windows drive-letter from the path for url like "c:\a\b".
            # As a workaround, we set subpath to the url. 
            subpath = url
```